### PR TITLE
Added rule to enfore stricter slicing operations and unit tests for it.

### PR DESCRIPTION
**Pull request by Eric4Jiang**
# I have made things!

## Checklist

- [x ] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ x] I have created at least one test case for the changes I have made
- [x ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ It features a lot of new rules from different categories.
 - Forbids to use and declare `float` keys in arrays and dictionaries
 - Forbids to use `a[len(a) - 1]` because it is just `a[-1]`
 - Forbids too long call chains like `foo(a)(b)(c)(d)`
+- Enforces stricter slice operation usage
```
Comment (sobolevn) [inline CHANGELOG.md]: This should go to `0.15.0` section

File: tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    '::3',
+    '4',
+    '3:7',
+    '3:7:2',
+    '3:',
+    ':7',
+    '3::2',
+    '5:7:',
+    ':7:2',
+    '3::',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py]: Looks like this subscript also does not make any sense. Why would anyone want to write `3::` and not just `3:`?

File: tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    ':7:',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py]: The same here. Why not just `:7`?

File: wemake_python_styleguide/violations/best_practices.py [hunk 4]
```diff
@@ -1941,3 +1943,30 @@ class FloatKeyViolation(ASTViolation):
...
+
+    Example::
+
+        # Correct:
+        items.reverse()
+
+        # Wrong:
+        items[::-1]
+
+    .. versionadded:: 0.13.0
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py]: .. versionadded:: 0.15.0

File: wemake_python_styleguide/violations/best_practices.py [hunk 5]
```diff
@@ -1941,3 +1943,30 @@ class FloatKeyViolation(ASTViolation):
...
+
+    """
+
+    error_template = 'Found undescriptive slice operation'
+    code = 500
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py]: Code is not correct. This violation should be located in `refactoring.py` and have a correct code.

File: wemake_python_styleguide/visitors/ast/subscripts.py [hunk 6]
```diff
@@ -23,9 +23,10 @@ def visit_Subscript(self, node: ast.Subscript) -> None:
 
         Raises:
             RedundantSubscriptViolation
-
+            UndescriptiveSliceOperationViolation
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/subscripts.py]: Missing new line after this

File: tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py (hunk 2 above)
Comment (sobolevn) [reply tests/test_visitors/test_ast/test_subscripts/test_undescriptive_slicer_operation.py]: Related https://github.com/wemake-services/wemake-python-styleguide/issues/1071

File: CHANGELOG.md (hunk 1 above)
Comment (Eric4Jiang) [reply CHANGELOG.md]: Could you be more specific? I don't see a section ```0.15.0``` in ```CHANGELOG.md```. It only goes up to ```0.13.3```

Comment (sobolevn) [reply CHANGELOG.md]: That's what you have to create 🙂

Comment (Eric4Jiang) [reply CHANGELOG.md]: Okay! Could you explain reasoning why this is going into a new section rather than 0.13.0?

Comment (sobolevn) [reply CHANGELOG.md]: Because `0.13` is already released, and we are working on `0.15`

**Comment by Eric4Jiang** [comment]
For e2e testing, what should my WPS code be? In the docs, I see naming, complexity, consistency, and best_practices, but no WPS range for refactoring.

**Comment by Eric4Jiang** [comment]
@sobolevn Just wanted to bump this in case you didn't see my last question.

**Comment by sobolevn** [comment]
@Eric4Jiang oh, sorry. I have missed it. The rage for refactoring is `5xx`. Can you please add it to the docs as well?

**Comment by sobolevn** [comment]
I am closing this for now, feel free to reopen if you still want to work on this. Cheers! 👍

**Pull request closed without merging**